### DISCUSSION TITLE
Let's encrypt acme challenge

### DISCUSSION THREAD

**Discussion by RizqiSyahrendra** [0xJacky/nginx-ui#174] [Q&A] [answered]
Hello

Let's encrypt requires me to add a file containing a unique code for acme challenge,  can you explain how to do that since I read that there's a 9180 as HTTPChallengePort, but I have no idea how to add a file there.

Thank you

**Comment by 0xJacky** [comment]
Try to add the following content to your configuration.
```
location /.well-known/acme-challenge {
        proxy_set_header Host $host;
        proxy_set_header X-Real_IP $remote_addr;
        proxy_set_header X-Forwarded-For $remote_addr:$remote_port;
        proxy_pass http://127.0.0.1:9180;
}
```

**Reply by RizqiSyahrendra** [reply]
Then where to place the unique code? let's encrypt will access http://[domain]/.well-known/acme-challenge/[unique code], right?

**Reply by 0xJacky** [reply]
Then you don't need do any thing, just turn on the switch, and nginx-ui will automatically apply the certification from let's encrypted for your sites.
<img width="319" alt="image" src="https://github.com/0xJacky/nginx-ui/assets/13096985/ed68c305-eade-47e6-a55b-b26d964b8a9a">

**Reply by RizqiSyahrendra** [reply]
Sorry, may I know where is that toggle button placed? I can't find it

**Reply by 0xJacky** [reply, marked as answer]
If you are adding a site using Nginx UI, you can find this button in the second step of configuring SSL.

If you are managing an existing site, please ensure that the "ssl" parameter is present in the listen directive of the server configuration block, so you can see this button.

**Reply by RizqiSyahrendra** [reply]
it works, thank you
